server/src/file_service.rs

```diff
@@ -33,10 +34,21 @@ pub async fn change_document_content(
     .await;
 
     let (old_content_version, new_version) = result.map_err(|e| match e {
-        file_index_repo::FileError::DoesNotExist => ChangeDocumentContentError::DocumentNotFound,
-        file_index_repo::FileError::IncorrectOldVersion => ChangeDocumentContentError::EditConflict,
-        file_index_repo::FileError::Deleted => ChangeDocumentContentError::DocumentDeleted,
-        _ => {
+        FileError::DoesNotExist => ChangeDocumentContentError::DocumentNotFound,
+        FileError::IncorrectOldVersion => ChangeDocumentContentError::EditConflict,
+        FileError::Deleted => ChangeDocumentContentError::DocumentDeleted,
+        FileError::FolderMovedIntoDescendants
```

Comment:
@tvanderstad / @SmailBarkouch are all of these supposed to be 500's? `ParentDoesNotExist` for example? Is there a way to model this so that `FileError`s aren't being grouped like this?

Reply:
I forgot to ask about this during stand-up, i'm confused on what you mean. Can you elaborate?